mfr/extensions/office365/render.py

```diff
@@ -0,0 +1,36 @@
+import os
+import furl
+
+from mfr.core import extension
+from mako.lookup import TemplateLookup
+from mfr.extensions.office365 import settings
+
+
+class Office365Renderer(extension.BaseRenderer):
+    """A renderer for use with public .docx files.
+
+    Office online can render .docx files to pdf for us.
+    This renderer will only ever be made if a query param with `public_file=1` is sent.
+    It then generates and embeds an office online url into an
+    iframe and returns the template. The file it is trying to render MUST
+    be available publically online. This renderer will not work if testing locally.
+
+    """
+
+    TEMPLATE = TemplateLookup(
+        directories=[
+            os.path.join(os.path.dirname(__file__), 'templates')
+        ]).get_template('viewer.mako')
+
+    def render(self):
+        download_url = furl.furl(self.metadata.download_url).set(query='')
+        url = settings.OFFICE_BASE_URL + download_url.url
```

Comment:
Encode the URL since it is the value of the `src` query param. It works without encoding during local test since the URL is probably not ambiguous.

Reply:
Can you try a revision download URL to see if it still works without encoding?

Reply:
- [ ] Need proper encoding.

What we feed to MS Office Live is an WB URL and here is an example of my local one which is more complicated than OSF url we test.

```
http://192.168.168.167:7777/v1/resources/ygcb4/providers/osfstorage/5a01dd60005aca00414c567d?action=download&public_file=1&version=1&mode=render&direct
```

Reply:
Here is what mine looks like
This is the `download_url` after the query params are stripped off
```
http://192.168.168.167:7777/v1/resources/3uq58/providers/osfstorage/5a03579db4eef2001827a8c3
```
as long as the project is public, going to that link will download the file. Also there are no chars that -need- to be encoded. If they should be encoded, well probably?  We briefly discussed this. Thoughts on encoding the `/` marks?